solvation_analysis/analysis.py

```diff
@@ -0,0 +1,342 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.analysis.rdf import InterRDF
+from MDAnalysis.lib.distances import capped_distance
+import numpy as np
+from solvation_analysis.rdf_parser import identify_solvation_cutoff
+from solvation_analysis.analysis_library import (
+    Coordination,
+    Pairing,
+    Speciation,
+)
+from solvation_analysis.solvation import get_radial_shell, get_closest_n_mol, get_atom_group
+
+
+class Solution(AnalysisBase):
+    """
+    The core class of the solvation module.
+
+    Parameters
+    ----------
+        solute : AtomGroup
+            the solute in the solutions
+        solvents: dict
+            a dictionary of names and atom groups. e.g. {"name_1": solvent_group_1,
+            "name_2": solvent_group_2, ...}
+        radii : dict, optional
+            an optional dictionary of solvation radii, any radii not
+            given will be calculated. e.g. {"name_2": radius_2, "name_5": radius_5}
+        rdf_kernel : function, optional
+            this function must take rdf bins and data as input and return
+            a solvation radius as output. e.g. rdf_kernel(bins, data) -> 3.2. By default,
+            the rdf_kernel is solvation_analysis.rdf_parser.identify_solvation_cutoff.
+        kernel_kwargs : dict, optional
+            kwargs passed to rdf_kernel
+        rdf_init_kwargs : dict, optional
+            kwargs passed to inner rdf initialization
+        rdf_run_kwargs : dict, optional
+            kwargs passed to inner rdf run e.g. inner_rdf.run(**rdf_run_kwargs)
+        kwargs : dict, optional
+            kwargs passed to AnalysisBase
+
+    Attributes
+    ----------
+    u : Universe
+        An MDAnalysis Universe object
+    n_solute : int
+        number of solute atoms
+    radii : dict
+        a dictionary of solvation radii for each solvent
+        e.g. {"name_2": radius_2, "name_2": radius_2, ...}
+    rdf_data : dict
+        a dictionary of rdf data, keys are solvent names and values
+        are (bins, data) tuples.
+    solvation_data : pandas.DataFrame
+        a dataframe of solvation data with columns "frame", "solvated_atom", "atom_id",
+        "dist", "res_name", and "res_id". If multiple entries share a frame, solvated_atom,
+        and atom_id, all but the the closest atom is dropped.
+    solvation_data_dup : pandas.DataFrame
```

Comment:
It would be useful if a user wanted to distinguish between monovalent and divalent solvents. Not sure if that warrants being included as a public attribute.